src/Servers/Kestrel/Transport.Sockets/src/SocketTransportFactory.cs

```diff
@@ -40,4 +41,16 @@ public ValueTask<IConnectionListener> BindAsync(EndPoint endpoint, CancellationT
         transport.Bind();
         return new ValueTask<IConnectionListener>(transport);
     }
+
+    /// <inheritdoc />
+    public bool CanBind(EndPoint endpoint)
```

Comment:
This could be an explicit interface implementation if we don't want it public.

Reply:
> Does anyone do this? Probably not. If someone is using a custom endpoint type to pass in custom values to the factory method, a workaround to this change is to inherit from one of the supported endpoint types, e.g. `IPEndPoint`, and add new members..

Would another workaround be to make CanBind virtual so they can derive from the transport and override it?

Reply:
Implementing a new interface on a public type is an API change with or without an explicit interface implementation. It doesn't matter what our PublicAPI.Unshipped.txt files have to say about it.

Personally, I think we should make `CanBind` fully public. It's not like it adds a lot of clutter to the AIP surface of `SocketTransportFactory`.

Reply:
Sure.

I don't like making `CanBind `virtual. Nothing else on this type is designed to be extended with inheritance. For example, the create socket factory method is configured via options as a func. If we want to allow people to customize `CanBind`, then it should also be a func on the options.

Reply:
I opened #45035 so we can discuss it in API review. I like it though.